iquip/apps/oven.py

```diff
@@ -0,0 +1,521 @@
+"""Module for oven controller."""
+
+from __future__ import annotations
+
+import functools
+import logging
+from typing import Any, Callable, List, Optional, Tuple
+
+from PyQt5.QtCore import pyqtSignal, pyqtSlot, QObject, Qt, QThread, QTimer
+from PyQt5.QtWidgets import (
+    QAbstractSpinBox, QDoubleSpinBox, QHBoxLayout, QPushButton, QSpinBox, QVBoxLayout, QWidget
+)
+
+import qiwis
+from sipyco.pc_rpc import Client
```

Comment:
`sipyco` is in an equivalence class with `PyQt5`, as it is an external 3rd-party package.